src/open_inwoner/utils/url.py

```diff
@@ -58,3 +61,23 @@ def prepend_next_url_param(url: str, next_url: str) -> str:
         next_url = prepend_next_url_param(next_url, current_next)
     u.args.set("next", next_url)
     return u.url
+
+
+def uuid_from_url(
+    url: str, allow_multiple: bool = False
+) -> uuid.UUID | list[uuid.UUID] | None:
+    """Extract UUID(s) from a given URL"""
+
+    pattern = r"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+    matches = re.findall(pattern, url, re.IGNORECASE)
+
+    if not matches:
+        return None
+    if len(matches) > 1 and allow_multiple:
+        return [uuid.UUID(match) for match in matches]
+    elif len(matches) > 1:
+        raise ValueError(
+            "url contains more than 1 UUID (use allow_multiple=True if you expect multiple results)"
+        )
```

Comment:
The cases seem a bit confusing to me, as both branches are special cases if `len(matches) > 1`:

```suggestion
    if len(matches) > 1:
        if allow_multiple:
          return [uuid.UUID(match) for match in matches]

        raise ValueError(
            "url contains more than 1 UUID (use allow_multiple=True if you expect multiple results)"
        )
```